### ISSUE TITLE
Kohya_ss LoRA Training Error: No Data Found Despite Correct Folder Setup

### ISSUE THREAD

**Issue by 001satyaki** [bmaltais/kohya_ss#3181]
Hi, I’ve been trying to train a LoRA model using Kohya_ss on Windows 11 with a GTX 1650 (4GB VRAM). My image folder structure is:

D:\AI\kohya_ss\training\goth_model\images\goth\ (14 JPGs inside)

In the Kohya GUI, I’ve set the image folder to:

D:\AI\kohya_ss\training\goth_model\images

But I keep getting this error:

ERROR: No data found. Please verify arguments (train_data_dir must be the parent of folders with images)

I’ve already resized images to 512x512, enabled xformers, and set correct resolution/batch size.

If anyone has solved this or can help debug, I’d really appreciate it. 🙏

Thank you in advance.                                                                                                                                                                                                                            And these are all the errors after a lot of trying                                                                                                                                                                                        

10:00:12-768676 INFO     Training has ended.
10:03:24-336396 INFO     Start training LoRA Standard ...
10:03:24-337396 INFO     Validating lr scheduler arguments...
10:03:24-339472 INFO     Validating optimizer arguments...
10:03:24-340461 INFO     Validating D:/AI/kohya_ss/output existence and writability... SUCCESS
10:03:24-341460 INFO     Validating runwayml/stable-diffusion-v1-5 existence... SKIPPING: huggingface.co model
10:03:24-343508 INFO     Validating D:/AI/kohya_ss/training/goth_model/images/goth existence... SUCCESS
10:03:24-345023 INFO     Regularization factor: 1
10:03:24-346022 INFO     Train batch size: 1
10:03:24-347228 INFO     Gradient accumulation steps: 1
10:03:24-347228 INFO     Epoch: 20
10:03:24-348647 INFO     Max train steps: 1600
10:03:24-349656 INFO     stop_text_encoder_training = 0
10:03:24-350662 INFO     lr_warmup_steps = 0.1
10:03:24-353894 INFO     Saving training config to D:/AI/kohya_ss/output\goth_lora_20250416-100324.json...
10:03:24-357573 INFO     Executing command: D:\AI\kohya_ss\venv\Scripts\accelerate.EXE launch --dynamo_backend no --dynamo_mode default --mixed_precision
                         fp16 --num_processes 1 --num_machines 1 --num_cpu_threads_per_process 2 D:/AI/kohya_ss/sd-scripts/train_network.py --config_file
                         D:/AI/kohya_ss/output/config_lora-20250416-100324.toml
2025-04-16 10:03:39 INFO     Loading settings from D:/AI/kohya_ss/output/config_lora-20250416-100324.toml...                              train_util.py:4621
2025-04-16 10:03:39 INFO     Using v1 tokenizer                                                                                            strategy_sd.py:26
D:\AI\kohya_ss\venv\Lib\site-packages\transformers\tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884
  warnings.warn(
2025-04-16 10:03:40 INFO     Using DreamBooth method.                                                                                   train_network.py:499
                    INFO     prepare images.                                                                                              train_util.py:2049
                    INFO     0 train images with repeats.                                                                                 train_util.py:2092
                    INFO     0 reg images with repeats.                                                                                   train_util.py:2096
                    WARNING  no regularization images / 正則化画像が見つかりませんでした                                                  train_util.py:2101
                    INFO     [Dataset 0]                                                                                                  config_util.py:575
                               batch_size: 1
                               resolution: (512, 512)
                               enable_bucket: False


                    INFO     [Prepare dataset 0]                                                                                          config_util.py:587
                    INFO     loading image sizes.                                                                                          train_util.py:970
0it [00:00, ?it/s]
                    INFO     make buckets                                                                                                  train_util.py:993
                    WARNING  min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is set, because bucket reso is defined  train_util.py:1010
                             by image size automatically /
                             bucket_no_upscaleが指定された場合は、bucketの解像度は画像サイズから自動計算されるため、min_bucket_resoとmax_
                             bucket_resoは無視されます
                    INFO     number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）                              train_util.py:1039
D:\AI\kohya_ss\venv\Lib\site-packages\numpy\core\fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
D:\AI\kohya_ss\venv\Lib\site-packages\numpy\core\_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
                    INFO     mean ar error (without repeats): nan                                                                         train_util.py:1049
                    ERROR    No data found. Please verify arguments (train_data_dir must be the parent of folders with images) /        train_network.py:545
                             画像がありません。引数指定を確認してください（train_data_dirには画像があるフォルダではなく、画像があるフォ
                             ルダの親フォルダを指定する必要があります）
10:03:42-665676 INFO     Training has ended.
10:19:31-067793 INFO     Start training LoRA Standard ...
10:19:31-068793 INFO     Validating lr scheduler arguments...
10:19:31-079370 INFO     Validating optimizer arguments...
10:19:31-080744 INFO     Validating D:/AI/kohya_ss/output existence and writability... SUCCESS
10:19:31-082175 INFO     Validating runwayml/stable-diffusion-v1-5 existence... SKIPPING: huggingface.co model
10:19:31-083657 INFO     Validating D:/AI/kohya_ss/training/goth_model/images existence... SUCCESS
10:19:31-084795 INFO     Error: 'goth' does not contain an underscore, skipping...
10:19:31-086210 INFO     Regularization factor: 1
10:19:31-087600 INFO     Train batch size: 1
10:19:31-088607 INFO     Gradient accumulation steps: 1
10:19:31-090010 INFO     Epoch: 20
10:19:31-091099 INFO     Max train steps: 1600
10:19:31-092107 INFO     stop_text_encoder_training = 0
10:19:31-092107 INFO     lr_warmup_steps = 0.1
10:19:31-095778 INFO     Saving training config to D:/AI/kohya_ss/output\goth_lora_20250416-101931.json...
10:19:31-100075 INFO     Executing command: D:\AI\kohya_ss\venv\Scripts\accelerate.EXE launch --dynamo_backend no --dynamo_mode default --mixed_precision
                         fp16 --num_processes 1 --num_machines 1 --num_cpu_threads_per_process 2 D:/AI/kohya_ss/sd-scripts/train_network.py --config_file
                         D:/AI/kohya_ss/output/config_lora-20250416-101931.toml
2025-04-16 10:19:47 INFO     Loading settings from D:/AI/kohya_ss/output/config_lora-20250416-101931.toml...                              train_util.py:4621
2025-04-16 10:19:47 INFO     Using v1 tokenizer                                                                                            strategy_sd.py:26
D:\AI\kohya_ss\venv\Lib\site-packages\transformers\tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884
  warnings.warn(
                    INFO     Using DreamBooth method.                                                                                   train_network.py:499
                    WARNING  ignore directory without repeats / 繰り返し回数のないディレクトリを無視します: goth                          config_util.py:608
                    INFO     prepare images.                                                                                              train_util.py:2049
                    INFO     0 train images with repeats.                                                                                 train_util.py:2092
                    INFO     0 reg images with repeats.                                                                                   train_util.py:2096
                    WARNING  no regularization images / 正則化画像が見つかりませんでした                                                  train_util.py:2101
                    INFO     [Dataset 0]                                                                                                  config_util.py:575
                               batch_size: 1
                               resolution: (512, 512)
                               enable_bucket: False


                    INFO     [Prepare dataset 0]                                                                                          config_util.py:587
                    INFO     loading image sizes.                                                                                          train_util.py:970
0it [00:00, ?it/s]
                    INFO     make buckets                                                                                                  train_util.py:993
                    WARNING  min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is set, because bucket reso is defined  train_util.py:1010
                             by image size automatically /
                             bucket_no_upscaleが指定された場合は、bucketの解像度は画像サイズから自動計算されるため、min_bucket_resoとmax_
                             bucket_resoは無視されます
                    INFO     number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）                              train_util.py:1039
D:\AI\kohya_ss\venv\Lib\site-packages\numpy\core\fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
D:\AI\kohya_ss\venv\Lib\site-packages\numpy\core\_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
                    INFO     mean ar error (without repeats): nan                                                                         train_util.py:1049
                    ERROR    No data found. Please verify arguments (train_data_dir must be the parent of folders with images) /        train_network.py:545
                             画像がありません。引数指定を確認してください（train_data_dirには画像があるフォルダではなく、画像があるフォ
                             ルダの親フォルダを指定する必要があります）
10:19:50-393422 INFO     Training has ended.
10:21:25-279717 INFO     Start training LoRA Standard ...
10:21:25-282041 INFO     Validating lr scheduler arguments...
10:21:25-283121 INFO     Validating optimizer arguments...
10:21:25-284123 INFO     Validating D:/AI/kohya_ss/output existence and writability... SUCCESS
10:21:25-286123 INFO     Validating runwayml/stable-diffusion-v1-5 existence... SKIPPING: huggingface.co model
10:21:25-287632 INFO     Validating D:/AI/kohya_ss/training/goth_model existence... SUCCESS
10:21:25-289645 INFO     Error: 'images' does not contain an underscore, skipping...
10:21:25-290645 INFO     Regularization factor: 1
10:21:25-291645 INFO     Train batch size: 1
10:21:25-292645 INFO     Gradient accumulation steps: 1
10:21:25-294645 INFO     Epoch: 20
10:21:25-295645 INFO     Max train steps: 1600
10:21:25-297645 INFO     stop_text_encoder_training = 0
10:21:25-298645 INFO     lr_warmup_steps = 0.1
10:21:25-301650 INFO     Saving training config to D:/AI/kohya_ss/output\goth_lora_20250416-102125.json...
10:21:25-304754 INFO     Executing command: D:\AI\kohya_ss\venv\Scripts\accelerate.EXE launch --dynamo_backend no --dynamo_mode default --mixed_precision
                         fp16 --num_processes 1 --num_machines 1 --num_cpu_threads_per_process 2 D:/AI/kohya_ss/sd-scripts/train_network.py --config_file
                         D:/AI/kohya_ss/output/config_lora-20250416-102125.toml
2025-04-16 10:22:11 INFO     Loading settings from D:/AI/kohya_ss/output/config_lora-20250416-102125.toml...                              train_util.py:4621
2025-04-16 10:22:11 INFO     Using v1 tokenizer                                                                                            strategy_sd.py:26
D:\AI\kohya_ss\venv\Lib\site-packages\transformers\tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884
  warnings.warn(
2025-04-16 10:22:12 INFO     Using DreamBooth method.                                                                                   train_network.py:499
                    WARNING  ignore directory without repeats / 繰り返し回数のないディレクトリを無視します: images                        config_util.py:608
                    INFO     prepare images.                                                                                              train_util.py:2049
                    INFO     0 train images with repeats.                                                                                 train_util.py:2092
                    INFO     0 reg images with repeats.                                                                                   train_util.py:2096
                    WARNING  no regularization images / 正則化画像が見つかりませんでした                                                  train_util.py:2101
                    INFO     [Dataset 0]                                                                                                  config_util.py:575
                               batch_size: 1
                               resolution: (512, 512)
                               enable_bucket: False


                    INFO     [Prepare dataset 0]                                                                                          config_util.py:587
                    INFO     loading image sizes.                                                                                          train_util.py:970
0it [00:00, ?it/s]
                    INFO     make buckets                                                                                                  train_util.py:993
                    WARNING  min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is set, because bucket reso is defined  train_util.py:1010
                             by image size automatically /
                             bucket_no_upscaleが指定された場合は、bucketの解像度は画像サイズから自動計算されるため、min_bucket_resoとmax_
                             bucket_resoは無視されます
                    INFO     number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）                              train_util.py:1039
D:\AI\kohya_ss\venv\Lib\site-packages\numpy\core\fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
D:\AI\kohya_ss\venv\Lib\site-packages\numpy\core\_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
                    INFO     mean ar error (without repeats): nan                                                                         train_util.py:1049
                    ERROR    No data found. Please verify arguments (train_data_dir must be the parent of folders with images) /        train_network.py:545
                             画像がありません。引数指定を確認してください（train_data_dirには画像があるフォルダではなく、画像があるフォ
                             ルダの親フォルダを指定する必要があります）
10:22:14-790982 INFO     Training has ended.

**Comment by kli017** [comment]
same error, any solution yet?

**Comment by 001satyaki** [comment]
> same error, any solution yet?

no dude! still stuck here. My brain is melting at this point.

**Comment by jordan-dalby** [comment]
@001satyaki @kli017 looking at the code, 
```
for folder in subfolders:
            try:
                # Extract the number of repeats from the folder name
                repeats = int(folder.split("_")[0])
```
The expected folder structure is:
```
training_data
   -> 001_data
   -> 002_data
   ...
```
I've currently got it running with just 001_data, no need for the second one I suppose? Then pass /path/to/training_data as the training data folder.

**Comment by 001satyaki** [comment]
@jordan-dalby Thanks, I have finally got it running! Thank you for the help.

**Comment by bmaltais** [comment]
> *This was generated by AI during triage.*

## Duplicate

Closing as a **duplicate of #3374**.

### Why this is the same issue
The `No data found` / `train_data_dir must be the parent of folders with images` error almost always means one of:

1. **Image folder** points at the leaf folder that holds the images (`…/10_concept`) instead of the **parent** of the `N_name` subfolders (`…/img`).
2. Subfolders are not named with the required `N_concept` pattern (e.g. `goth` instead of `10_goth`).

Required layout (DreamBooth / LoRA / TI folder mode):

```text
train_data_dir/          ← set this path in the GUI
  10_my_concept/
    image1.jpg
    image1.txt
```

Docs: [`docs/image_folder_structure.md`](https://github.com/bmaltais/kohya_ss/blob/master/docs/image_folder_structure.md)

### What happens next
#3374 is kept open as the **canonical** ticket for a product fix: the GUI already has a `verify_image_folder_pattern` helper but it is never called on train start, so users only see the opaque error from sd-scripts. Tracking the implementation work there.
